flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointFailureManager.java

```diff
@@ -204,7 +204,8 @@ private void checkFailureAgainstCounter(
             if (continuousFailureCounter.get() > tolerableCpFailureNumber) {
                 clearCount();
                 errorHandler.accept(
-                        new FlinkRuntimeException(EXCEEDED_CHECKPOINT_TOLERABLE_FAILURE_MESSAGE));
+                        new FlinkRuntimeException(
+                                EXCEEDED_CHECKPOINT_TOLERABLE_FAILURE_MESSAGE, exception));
```

Comment:
The job failed due to the failure counter being larger than the tolerable number, and we can only have the exception reason for the last broken checkpoint. However, this would make users think all checkpoints failed due to the last exception. The correct way is to let users check the job manager logs or checkpoint UI to know what happened in the last checkpoints.
From my point of view, I am +0 for this proposal.

Reply:
@Myasuka Thanks for your feedback. 

You are right, the correct way is check full information from JM log or checkpoint UI. 

Actually, I added this due to some reasons:

- Some Flink platforms collect exceptions. When the job fails and JM stops, users can easily see the root cause of the last checkpoint through the exception. At this point WebUI has stopped, and it is more convenient than JM LOG.
- Displaying more root cause has no effect on the original logic.
- When developing some features, ITCase is often run without LOG enabled. Some ITCases fail, it just shows `Exceeded checkpoint tolerable failure threshold.`, doesn't show the root cause. Inconvenient to locate the problem. 😂

I also don't think this change is necessary. You can take a look at these reasons and I will close this PR if not needed. Thanks~

Reply:
I think we can add some descriptions to tell users to refer to the checkpoint history tab or job manager logs to see why continuous checkpoints failed. That could be better than the current hint.

Reply:
Thanks for your suggestion. I have added some descriptions.

Reply:
My previous suggestion hopes you could remove the `exception` from the `FlinkRuntimeException` to avoid misunderstanding for users.

Reply:
How about this? We tell user the latest checkpoint failed cause and how to check full checkpoint info?

The `exception.getCheckpointFailureReason().message()` is a short description.

```
public static final String EXCEEDED_CHECKPOINT_TOLERABLE_FAILURE_MESSAGE =
            "Exceeded checkpoint tolerable failure threshold, the latest checkpoint failed due to %s,"
                    + " view the Checkpoint History tab or the Job Manager log to find out why"
                    + " continuous checkpoints failed.";

new FlinkRuntimeException(String.format(
        EXCEEDED_CHECKPOINT_TOLERABLE_FAILURE_MESSAGE,
        exception.getCheckpointFailureReason().message()));
```

Reply:
I think this might be a better idea. BTW, could you please make `Exceeded checkpoint tolerable failure threshold` ends with a period instead of a comma to make it the same as before.

Reply:
Updated.